dev/buildtool/build_commands.py

```diff
@@ -0,0 +1,732 @@
+    ]
+
+    gradle_root = self.determine_gradle_root(repository)
+    logfile = os.path.join(self.options.scratch_dir, name,
+                           '{name}-docker-build.log'.format(name=name))
+    check_subprocesses_to_logfile(
+        '{name} docker build'.format(name=name), logfile,
+        cmds, cwd=gradle_root)
+
+  def __build_with_gcb(self, repository):
+    name = repository.name
+    version = self._get_nebula_repository_version(name)
+    gcb_config = self.__derive_gcb_config(name, version)
+    if gcb_config is None:
+      logging.info('Skipping GCB for %s because there is config for it',
+                   name)
+      return
+
+    options = self.options
+    log_flags = '--log-http' if options.log_level == 'debug' else ''
+    name_scratch_dir = os.path.join(options.scratch_dir, name)
+
+    # Use an absoluate path here because we're going to
+    # pass this to the gcloud command, which will be running
+    # in a different directory so relative paths wont hold.
+    config_path = os.path.abspath(os.path.join(
+        name_scratch_dir, '{name}-gcb.yml'.format(name=name)))
+    write_to_path(gcb_config, config_path)
+
+    # NOTE(ewiseblatt): 20171213
+    # This is an old comment. Not sure if it is still relevant or not.
+    #
+    # Local .gradle dir stomps on GCB's .gradle directory when the gradle
+    # wrapper is installed, so we need to delete the local one.
+    # The .gradle dir is transient and will be recreated on the next gradle
+    # build, so this is OK.
+    nebula_dir = self._get_nebula_repository_path(name)
+
+    # This can still be shared among components as long as the
+    # scratch directory remains around.
+    if options.force_clean_gradle_cache:
+      # If we're going to delete existing ones, then keep each component
+      # separate so they dont stomp on one another
+      gradle_cache = os.path.abspath(os.path.join(nebula_dir, '.gradle'))
+    else:
+      # Otherwise allow all the components to share a common gradle directory
+      gradle_cache = os.path.abspath(
+          os.path.join(options.scratch_dir, '.gradle'))
+
+    if options.force_clean_gradle_cache and os.path.isdir(gradle_cache):
+      shutil.rmtree(gradle_cache)
+
+    # Note this command assumes a cwd of nebula_dir
+    cmd = ('gcloud container builds submit {log_flags}'
+           ' --account={account} --project={project}'
+           ' --config="{config_path}" .'
+           .format(log_flags=log_flags,
+                   account=options.gcb_service_account,
+                   project=options.gcb_project,
+                   config_path=config_path))
+
+    logfile = os.path.join(
+        name_scratch_dir, '{name}-gcb-build.log'.format(name=name))
+    check_subprocesses_to_logfile(
+        '{name} container build'.format(name=name), logfile,
+        [cmd], cwd=nebula_dir)
+
+  def __make_gradle_gcb_step(self, name, env_vars_list):
+    # Nebula complains if the git tag version doesn't match the branch's
```

Comment:
This comment is describing a `git checkout` but it doesn't appear in this block of code.

Reply:
oops